### ISSUE TITLE
Multiple file select - builtin file chooser

### ISSUE THREAD

**Issue by jonnieey** [atlas-engineer/nyxt#3069]
I cannot attach multiple files at once using the built in file chooser

use regex to attach. Ex, ``/home/user/docs/*.pdf`` to attach all pdfs in the directory

Selecting file one by one

**Comment by jmercouris** [comment]
Hello, thank you for your report, what happens if you press `M-a` within the built in selection dialog, does that work for you? Can you specify which command you are using the file chooser with?

**Comment by jonnieey** [comment]
> Hello, thank you for your report, what happens if you press `M-a` within the built in selection dialog, does that work for you? Can you specify which command you are using the file chooser with?

so i want to upload a file to a website, i press the upload button and the `file to input` prompt appears. The command used is default, i guess.

I'm using vi-mode too.

`M-a`, `M-m` or `M-u` don't do anything

**Comment by jmercouris** [comment]
I see, I think you can only upload a single file at a time in those forms. I don't think they select multiple files, I could be wrong though.

**Comment by jonnieey** [comment]
> I see, I think you can only upload a single file at a time in those forms. I don't think they select multiple files, I could be wrong though.

Yeah, just tested on another website and `M-a` works. To filter you just have to type extension and press `M-a`. `/home/user/docs/pdf` then press `M-a`

**Comment by jonnieey** [comment]
> @jmercouris it should be possible. See `process-file-chooser-request`.
> 
> @jonnieey could you provide a recipe to reproduce this issue? Thanks.

conversing on google chat. Needed to send some photos but could only do one at a time. When sending an email you can select multiple files.

**Comment by jonnieey** [comment]
> It seems that everything's now clear.
> 
> @jmercouris should we keep this open in case someone wants to implement regex matching for the prompt buffer? I'd rather close it perhaps.

Just to mention, in other browsers (qutebrowser) it's possible to select multiple files. On android too.

**Comment by jonnieey** [comment]
> To figure out the issue I need to be able to reproduce it. Unfortunately I don't user google chat. 
> 
> Another way you could help to trace the issue would be to see if you can reproduce it on Gnome Web (also known as epiphany) or the surf browser. This would let us understand whether the issue is due to us or the renderer.

Tested on surf, i get same behaviour. Cannot select multiple on that upload field. Guess it's the renderer then.

**Comment by aartaka** [comment]
> Tested on surf, i get same behaviour. Cannot select multiple on that upload field. Guess it's the renderer then.

It's might not even be the renderer—some file inputs only allow one file to upload.

**Comment by aartaka** [comment]
We are basically following the data that renderer gives us, and the renderer is basically choosing singe/multiple inputs based on what the file input allows.
